### PR TITLE
update FE to account for new dateOfContact field

### PR DESCRIPTION
**Pull request by StuTruss**
## Summary

The purpose of this PR is to show the new date of contact field in the service items details section alongside the `timeMilitary` field. Also updates how we render `First available dates` and `customer contact attempts` to account for the case where there may be more than 2 of each. 

Check out [this slack thread if you would like extra context](https://ustcdp3.slack.com/archives/CTNTFJSBA/p1686772959097629)

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Run `make server_test`
2. All tests should pass
3. Create a new customer `HHG` move and move it along the system to make it available to the prime
4. Login into to the Prime UI and access your newly created move
5. Click the `Add Service Item` button
6. Scroll down to the form and select `Destination SIT` from the `Service item type` dropdown
7. The `date of attempted contact` field should be present - select a date, complete the form, and submit
8. It should now be reflected in the `mto_service_item_customer_contacts` table if you check your DB
9. You can also access the move as a `TOO` and check the move task order tab to check the service items section 

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/about/supported-browsers) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/about/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Service Item Section
Screenshot 2023-06-20 at 3 33 21 PM

### CODE REVIEW

**Review by ruizajtruss — APPROVED** [review]
Code is clean and straightforward, LGTM 🚀  Thanks for working on this!

File: src/components/Office/ServiceItemDetails/ServiceItemDetails.jsx [hunk 1]
```diff
@@ -44,35 +45,47 @@ const ServiceItemDetails = ({ id, code, details }) => {
...
+      const sortedDeliveryDates = sortBy(customerContacts, [
+        (a) => {
+          return new Date(a.firstAvailableDeliveryDate);
+        },
+      ]);
+      const sortedDateOfContacts = sortBy(customerContacts, [
+        (a) => {
+          return new Date(a.dateOfContact);
+        },
+      ]);
```
Comment (afoon) [inline src/components/Office/ServiceItemDetails/ServiceItemDetails.jsx:58]: **suggestion**: Can we add some comments or JSdoc to explain what is going on here? I can see this being confusing for engineers not familiar with lodash.

**Comment by AnnaGingle** [comment]
First available delivery date 1 & Customer contact attempt 1 go together 
First available delivery date 1 & Customer contact attempt 1 are required but additional iterations are not (2,3,4)
SIT service items

**Review by AnnaGingle — CHANGES_REQUESTED** [review]
First available delivery date 1 & Customer contact attempt 1 go together
First available delivery date 1 & Customer contact attempt 1 are required but additional iterations are not (2,3,4)
SIT service items

**Comment by StuTruss** [comment]
@ruizajtruss @donaldthai could we get some clarity on what is expected here? It was my understanding that we wanted to change the FE to account for multiple `First available delivery date` as well as having `first available delivery date` & `customer contact` go together.

**Comment by StuTruss** [comment]
> @ruizajtruss @donaldthai could we get some clarity on what is expected here? It was my understanding that we wanted to change the FE to account for multiple `First available delivery date` as well as having `first available delivery date` & `customer contact` go together.

for context this was the image i received as far as what was the intended outcome of this PR 
image

**Comment by AnnaGingle** [comment]
@StuTruss There was an issue with updating our design system components. I can confirm this is [the expected design](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?type=design&node-id=13645-139219&mode=design&t=DqO0W3z7WbX0F8b5-4). Sorry for the confusion.
SIT service items

**Comment by AnnaGingle** [comment]
This is what I am seeing in the PR
Screenshot 2023-06-27 at 12 12 30 PM

**Comment by StuTruss** [comment]
thanks for the clarification! @AnnaGingle

**Comment by StuTruss** [comment]
> This is what I am seeing in the PR Screenshot 2023-06-27 at 12 12 30 PM

@AnnaGingle is this how it should look if there is no available date or customer contact? 
Screenshot 2023-06-27 at 11 11 21 AM

**Comment by AnnaGingle** [comment]
@StuTruss The empty states look right. We use the - dash.
Screenshot 2023-06-27 at 2 43 01 PM


Also, is there a testharness move with this info for me to check? I redeployed the PR & still am not seeing these fields.

**Comment by AnnaGingle** [comment]
I may have spoken too soon. Im checking the Happo report & the new time format isn't tracking here.
Screenshot 2023-06-28 at 9 33 11 AM
Screenshot 2023-06-28 at 9 34 08 AM

**Review by AnnaGingle — APPROVED** [review]
Approved! Thanks for your patience. I will follow up on the time format for the modal.

**Pull request merged**